core/src/main/scala/org/scalafmt/config/Newlines.scala

```diff
@@ -44,5 +62,18 @@ case class Newlines(
     sometimesBeforeColonInMethodReturnType: Boolean = true,
     penalizeSingleSelectMultiArgList: Boolean = true,
     alwaysBeforeCurlyBraceLambdaParams: Boolean = false,
-    alwaysBeforeTopLevelStatements: Boolean = false
+    alwaysBeforeTopLevelStatements: Boolean = false,
+    afterCurlyLambda: NewlineCurlyLambda = NewlineCurlyLambda.never
 )
+
+sealed abstract class NewlineCurlyLambda
+
+object NewlineCurlyLambda {
+
+  case object preserve extends NewlineCurlyLambda
+  case object always extends NewlineCurlyLambda
+  case object never extends NewlineCurlyLambda
+
+  implicit val newlineCurlyLambdaReader: Reader[NewlineCurlyLambda] =
```

Comment:
Good point. I agree that absolute readers like this one should exist in the companion object. Only the `@ConfigReader` generated readers are not implicit and need to be inside `ScalafmtConfig`. 

> due to a macro error I couldn't understand.

Yeah, metaconfig/macros can be tricky. I badly wanted to avoid rolling my own config library/macros but I couldn't find anything that checked all boxes at the time, see https://github.com/scalameta/scalafmt/issues/316 I think I might be able to use circe path readers, but I didn't know about them at the time.